code/modules/power/apc/apc_tool_act.dm

```diff
@@ -223,3 +223,5 @@
 				ui_interact(user)
 		else
 			balloon_alert(user, "access denied!")
+			if(isethereal(user))
+				sleep(1 SECONDS) // sleep here to prevent later balloon alerts that show up to ethereals from overlapping and making it completely unreadable
```

Comment:
I don't like the way this looks, and I also don't like that it presumably delays ethereal charging by a second. I think `togglelock` should return a value denoting that access was denied, then ethereals read that and delay just the balloon alert by a second with addtimer.

Reply:
the thing is that every single path in togglelock() will result in a balloon_alert that _will_ collide with anything in ethereal-specific interactions. I get what you mean by not wanting to delay ethereal charging by a second. I think I have an idea on this though.